### PR TITLE
DOC: Update the description of byteswap

### PR DESCRIPTION
**Pull request by kianasun**
The description did not mention complex numbers and strings (only appear below in the example)

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -2740,6 +2740,8 @@
 
     Toggle between low-endian and big-endian data representation by
     returning a byteswapped array, optionally swapped in-place.
+    Arrays of strings are not swapped. The real and imaginary parts
```
Comment (eric-wieser) [inline numpy/core/_add_newdocs.py]: This comment isn't true - bytes _are_ swapped for arrays of (unicode) strings:
```
>>> A = np.array(['ceg', 'fac'])
>>> A.view(np.uint8)
array([ 99,   0,   0,   0, 101,   0,   0,   0, 103,   0,   0,   0, 102,
         0,   0,   0,  97,   0,   0,   0,  99,   0,   0,   0], dtype=uint8)

>>> A.byteswap().view(np.uint8)
array([  0,   0,   0,  99,   0,   0,   0, 101,   0,   0,   0, 103,   0,
         0,   0, 102,   0,   0,   0,  97,   0,   0,   0,  99], dtype=uint8)

Comment (kianasun) [reply numpy/core/_add_newdocs.py]: You are right. Should we remove the comment about strings and the example below?

Comment (kianasun) [reply numpy/core/_add_newdocs.py]: The example is actually correct. It will raise `UnicodeDecodeError` when swapping arrays of strings.

Comment (eric-wieser) [reply numpy/core/_add_newdocs.py]: The example does not show what you think it does.
This works fine:
```
>>> A = np.array(['ceg', 'fac'])
>>> B = A.byteswap()  # no error
>>> print(B)  # UnicodeDecodeError
```

I think that example was trying to show what happens with _byte-strings_, so should have been `np.array([b'ceg', b'fac'])`

Comment (kianasun) [reply numpy/core/_add_newdocs.py]: I see. I can update the example to show what happens with byte-strings. Should I also add an example to show ``arr.newbyteorder().byteswap()`` here?

**Comment by kianasun** [comment]
@eric-wieser  I updated the example and added a new example to show ``arr.newbyteorder().byteswap()``.

**Comment by mattip** [comment]
Closes #13914

**Comment by mattip** [comment]
Thanks @kianasun, @eric-wieser.

**Pull request merged**